build_overrides/asr.gni

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+declare_args() {
+  # ASR IC family.
+  asr_ic_family = ""
+  # Root directory for ASR SDK build files.
+  asr_sdk_build_root = ""
+  # Root directory for ASR toolchain.
+  asr_toolchain_root = ""
+}
+
+if (asr_ic_family == "") {
+  asr_ic_family = getenv("ASR_IC")
+}
+
+assert(asr_ic_family != "", "ASR_IC must be specified")
+
+if (asr_ic_family == "asr582x") {
+  asr_sdk_build_root = "//third_party/connectedhomeip/third_party/asr/asr582x"
+}
+
+if (asr_ic_family == "asr595x") {
+  asr_sdk_build_root = "//third_party/connectedhomeip/third_party/asr/asr595x"
+}
+
+if (asr_toolchain_root == "") {
+  asr_toolchain_root = getenv("TOOLCHAIN_PATH")
```

Comment:
use ASR_TOOLCHAIN_PATH or similar as this is ASR specific. I imagine antyhing that is platform specific (TOOLCHAIN, SDK and similar) should likely have some platform-specific prefix.

Reply:
Thank you for your suggestion, we will change as soon as possible, and fully test before commit again!

Reply:
@andy31415  PR again:   https://github.com/project-chip/connectedhomeip/pull/26361